From dcca66140103855c8253a6e14924ef42592f580c Mon Sep 17 00:00:00 2001
From: Baihui Zeng <baihzeng@qti.qualcomm.com>
Date: Thu, 10 Jul 2025 10:40:30 +0800
Subject: [PATCH] asan-forbid-allocator64

Upstream-Status: Pending

Signed-off-by: Baihui Zeng <baihzeng@qti.qualcomm.com>
---
 libsanitizer/sanitizer_common/sanitizer_platform.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libsanitizer/sanitizer_common/sanitizer_platform.h b/libsanitizer/sanitizer_common/sanitizer_platform.h
index 7ecc465be..82fc4624f 100644
--- a/libsanitizer/sanitizer_common/sanitizer_platform.h
+++ b/libsanitizer/sanitizer_common/sanitizer_platform.h
@@ -287,7 +287,7 @@
 #  if (SANITIZER_ANDROID && defined(__aarch64__)) || SANITIZER_FUCHSIA
 #    define SANITIZER_CAN_USE_ALLOCATOR64 1
 #  elif defined(__mips64) || defined(__arm__) || defined(__i386__) || \
-      SANITIZER_RISCV64 || defined(__hexagon__)
+      SANITIZER_RISCV64 || defined(__hexagon__) || defined(__aarch64__)
 #    define SANITIZER_CAN_USE_ALLOCATOR64 0
 #  else
 #    define SANITIZER_CAN_USE_ALLOCATOR64 (SANITIZER_WORDSIZE == 64)
--
2.34.1

