From 831352dc8e6f85ba170507d22f47baf2b9fbbffe Mon Sep 17 00:00:00 2001
From: Lei wang <quic_leiwan@quicinc.com>
Date: Wed, 21 Aug 2024 09:47:14 +0800
Subject: [PATCH] udev-trigger: only enable must part while leave others late

Signed-off-by: Lei wang <quic_leiwan@quicinc.com>

Upstream-Status: Inappropriate [only for rumi case]

---
 units/systemd-udev-trigger.service | 1 -
 1 file changed, 1 deletion(-)

diff --git a/units/systemd-udev-trigger.service b/units/systemd-udev-trigger.service
index 18e1bc5c4b..2e31ca3ccd 100644
--- a/units/systemd-udev-trigger.service
+++ b/units/systemd-udev-trigger.service
@@ -19,4 +19,4 @@ ConditionPathIsReadWrite=/sys
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=-udevadm trigger --type=all --action=add --prioritized-subsystem=input,block,module,tpmrm,net,tty
+ExecStart=-udevadm trigger -w --subsystem-match=block --type=devices --action=add
-- 
2.34.1